src/main/java/net/neoforged/neoforge/attachment/IAttachmentHolder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.attachment;
+
+import java.util.function.Supplier;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An object that can hold data attachments.
+ */
+public interface IAttachmentHolder {
+    /**
+     * Returns {@code true} if there is a data attachment of the give type, {@code false} otherwise.
+     */
+    boolean hasData(AttachmentType<?> type);
+
+    /**
+     * Returns {@code true} if there is a data attachment of the give type, {@code false} otherwise.
+     */
+    default boolean hasData(Supplier<AttachmentType<?>> type) {
+        return hasData(type.get());
+    }
+
+    /**
+     * Returns the data attachment of the given type.
```

Comment:
This should use the inline `{@return}` tag.

(Same comment for this method's overload.)